arangod/Cluster/ActionBase.h

```diff
@@ -112,6 +112,14 @@ class ActionBase {
   /// @brief finalize statistics
   void endStats();
 
+  /**
+   *  @brief    update progress by long running processes, while allowing to
+   *            abort the process by user.
+   *  @param  u percentage of work done
+   *  @return   abort if !ok(), true if ok(), with reason to abort.
+   */
+  virtual arangodb::Result setProgress(uint64_t u);
```

Comment:
Maybe double progress is better?

Reply:
Maybe. This is not new code. I have no hard feelings either way. Out of scope of this PR.